case_utils/case_validate/__init__.py

```diff
@@ -64,13 +65,189 @@ class NonExistentCDOConceptWarning(UserWarning):
     pass
 
 
+class ValidationResult:
```

Comment:
I've been uncomfortable with the returned value of the `case_utils.validate()` function instead of something more parallel with `pyshacl.validate()`.  My main concern, which I think boils down to favoring non-invention, is that `case_utils.validate` would return something that is structurally different to the tune of "named tuple" vs. "unnamed tuple".

But, I think I ultimately agree with using a `ValidationResult` object.  Reasons:

* _Need_: `case_utils.validate` will return one additional "field" necessarily special to CDO, the set of typo'd concepts.  So already there must be a type-difference between `case_utils.validate` and `pyshacl.validate`.
* _User experience in upstream updates_*: Say we went with the unnamed-tuple pattern parallel with `pyshacl.validate`.  `pyshacl.validate` returns a 3-tuple, `case_utils.validate` returns a 4-tuple.  If `pyshacl.validate` decides down the road to return another field (& I could see that `Exception` going into an `Optional[Exception]` field), we're backed into a rougher corner in accommodating.  Returning a 5-tuple now causes potential for confusion with a downstream application using `pyshacl.validate(...)[3]`: Is that `pyshacl`'s (e.g. `Optional[Exception]`), or is it `case-utils`' (i.e. `Set[URIRef]`)?  Using the `ValidationResult` object, especially **not** as an ordered `NamedTuple`, saves on field confusion, because its results have to be accessed by name.  The day `pyshacl` releases a `validate()` function with a new return signature would induce a need for `case-utils` to change its call to `pyshacl.validate`, but no codebases downstream from `case-utils` would see an impact.

So, I'm fine with this object being defined and returned.  @kchason , if you agree, please feel free to Resolve this conversation.  I'll post the type updates from the other threads Monday.